daemon/graphdriver/graphtest/graphtest_unix.go

```diff
@@ -329,8 +332,14 @@ func DriverTestSetQuota(t *testing.T, drivername string) {
 
 	quota := uint64(50 * units.MiB)
 
-	err = writeRandomFile(path.Join(mountPath.Path(), "file"), quota*2)
-	if pathError, ok := err.(*os.PathError); ok && pathError.Err != unix.EDQUOT {
-		t.Fatalf("expect write() to fail with %v, got %v", unix.EDQUOT, err)
+	// Try to write a file smaller than quota, and ensure it works
+	err = writeRandomFile(path.Join(mountPath.Path(), "file"), quota/2)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Try to write a file bigger than quota, and ensure it does not work
+	if pathError, ok := err.(*os.PathError); ok && pathError.Err != unix.EDQUOT && pathError.Err != unix.ENOSPC {
+		t.Fatalf("expect write() to fail with %v or %v, got %v", unix.EDQUOT, unix.ENOSPC, err)
```

Comment:
Perhaps you want to print ``pathError.Err`` here, not ``err``, as this is what you use in comparisons?